### PR TITLE
add tmap (#15)

### PR DESCRIPTION
**Pull request by jw3126**

### CODE REVIEW

**Comment by jw3126** [comment]
Interesting, the `@inferred` all pass on my machine with `julia 1.1.0`. The CI fail is on a six argument function though, so maybe I should just reduce that number.  Also is support of `v0.6` still needed, or can I drop it?

**Comment by jw3126** [comment]
@mohamed82008 could you review?

**Comment by mohamed82008** [comment]
Yes sorry I was really busy the last week. I will try to review it tomorrow.

File: src/KissThreading.jl [hunk 1]
```diff
@@ -60,7 +60,7 @@ end
     end
 end
 
-function tmap!(f, dst::AbstractArray, src::AbstractArray...; batch_size=1)
+@noinline function tmap!(f, dst::AbstractArray, src::AbstractArray...; batch_size=1)
```
Comment (mohamed82008) [inline src/KissThreading.jl]: Why is this necessary?

File: src/KissThreading.jl [hunk 2]
```diff
@@ -70,6 +70,13 @@ function tmap!(f, dst::AbstractArray, src::AbstractArray...; batch_size=1)
     mapper(batch_size, f, dst, src...)
 end
 
+function tmap(f, src::AbstractArray...; batch_size=1)
+    g = Base.Generator(f,src...)
+    T = Base.@default_eltype(g)
```
Comment (mohamed82008) [inline src/KissThreading.jl]: Nice!

File: src/KissThreading.jl (hunk 1 above)
Comment (jw3126) [reply src/KissThreading.jl]: It is not, I removed it.

File: src/KissThreading.jl (hunk 2 above)
Comment (jw3126) [reply src/KissThreading.jl]: Thx!

**Comment by mohamed82008** [comment]
Thanks for the PR!
